### PR TITLE
UIDATIMP-1174: When user have Can view only permission, don't show Actions and +New button

### PR DESCRIPTION
**Pull request by Javokhir12**
## Purpose
- When user have `Can view only` permission, don't show Actions and +New buttons

## Approach
- Add checks to hide `Action menu` if user doesn't have correct permission

## Refs
- https://issues.folio.org/browse/UIDATIMP-1174

## Screenshots

https://user-images.githubusercontent.com/89512426/169525122-dbf5b222-7e3f-4e7c-ac83-bd99afbd09a0.mp4

### CODE REVIEW

**Review by OleksandrHladchenko1 — APPROVED** [review]

**Review by jakhongirabdukhamdiov — APPROVED** [review]

File: src/components/ListView/ListView.js [hunk 1]
```diff
@@ -19,16 +19,20 @@ import { buildUrl } from '@folio/stripes/smart-components';
...
+} from '../ViewContainer';
 import { SearchAndSort } from '../SearchAndSort';
-import { ViewContainer } from '../ViewContainer';
 import { listTemplate } from '../ListTemplate';
 import { ActionMenu } from '../ActionMenu';
 import { createNetworkMessage } from '../Callout';
 
 export class ListView extends Component {
   static propTypes = {
+    stripes: PropTypes.object,
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.js:35]: Use `stripesShape` from `@folio/stripes/core`

File: src/settings/ActionProfiles/ViewActionProfile.js [hunk 2]
```diff
@@ -71,6 +72,7 @@ export class ViewActionProfile extends Component {
   });
 
   static propTypes = {
+    stripes: PropTypes.object,
```
Comment (mariia-aloshyna) [inline src/settings/ActionProfiles/ViewActionProfile.js:75]: Same here

File: src/settings/ActionProfiles/ViewActionProfile.js [hunk 3]
```diff
@@ -157,7 +159,10 @@ export class ViewActionProfile extends Component {
   };
 
   renderPaneHeader = renderProps => {
-    const { onClose } = this.props;
+    const {
+      onClose,
+      stripes
```
Comment (mariia-aloshyna) [inline src/settings/ActionProfiles/ViewActionProfile.js:164]:
```suggestion
      stripes,
```

File: src/components/ListView/ListView.js [hunk 4]
```diff
@@ -241,7 +246,10 @@ export class ListView extends Component {
               resultCountMessageKey={`ui-data-import.settings.${ENTITY_KEY}.count`}
               resultsLabel={label}
               defaultSort={defaultSort}
-              actionMenu={actionMenu}
+              actionMenu={showActionMenu({
+                renderer: actionMenu,
+                stripes
```
Comment (mariia-aloshyna) [inline src/components/ListView/ListView.js:251]:
```suggestion
                stripes,
```

File: src/settings/FileExtensions/ViewFileExtension.js [hunk 5]
```diff
@@ -137,15 +139,21 @@ export class ViewFileExtension extends Component {
   );
 
   renderPaneHeader = renderProps => {
-    const { onClose } = this.props;
+    const {
+      onClose,
+      stripes
```
Comment (mariia-aloshyna) [inline src/settings/FileExtensions/ViewFileExtension.js:144]:
```suggestion
      stripes,
```

File: src/settings/JobProfiles/ViewJobProfile.js [hunk 6]
```diff
@@ -210,7 +212,10 @@ const ViewJobProfileComponent = props => {
         {...renderProps}
         paneTitle={paneTitle}
         paneSub={<FormattedMessage id="ui-data-import.jobProfileName" />}
-        actionMenu={actionMenu}
+        actionMenu={showActionMenu({
+          renderer: actionMenu,
+          stripes
```
Comment (mariia-aloshyna) [inline src/settings/JobProfiles/ViewJobProfile.js:217]:
```suggestion
          stripes,
```

File: src/settings/MARCFieldProtection/MARCFieldProtection.js [hunk 7]
```diff
@@ -18,7 +18,10 @@ import { withStripes } from '@folio/stripes/core';
 import { ControlledVocab } from '@folio/stripes/smart-components';
 import { TextField } from '@folio/stripes/components';
 
-import { MARC_FIELD_PROTECTION_SOURCE } from '../../utils';
+import {
+  MARC_FIELD_PROTECTION_SOURCE,
+  permissions
```
Comment (mariia-aloshyna) [inline src/settings/MARCFieldProtection/MARCFieldProtection.js:23]:
```suggestion
  permissions,
```

File: src/settings/MappingProfiles/ViewMappingProfile.js [hunk 8]
```diff
@@ -177,7 +179,10 @@ export class ViewMappingProfile extends Component {
   };
 
   renderPaneHeader = renderProps => {
-    const { onClose } = this.props;
+    const {
+      onClose,
+      stripes
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/ViewMappingProfile.js:184]:
```suggestion
      stripes,
```

File: src/settings/MappingProfiles/ViewMappingProfile.js [hunk 9]
```diff
@@ -196,7 +201,10 @@ export class ViewMappingProfile extends Component {
         {...renderProps}
         paneTitle={paneTitle}
         paneSub={<FormattedMessage id="ui-data-import.mappingProfileName" />}
-        actionMenu={this.renderActionMenu}
+        actionMenu={showActionMenu({
+          renderer: this.renderActionMenu,
+          stripes
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/ViewMappingProfile.js:206]:
```suggestion
          stripes,
```

File: src/utils/showActionMenu.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import { permissions } from './permissions';
+
+/**
+ * Displays Action menu if user has appropriate permission(s).
+ *
+ * @param {{renderer: Function | null, stripes: Object}} options
+*/
+export function showActionMenu({ renderer, stripes }) {
+  const hasPerm = stripes.hasPerm(permissions.SETTINGS_MANAGE);
+  return hasPerm ? renderer : null;
```
Comment (mariia-aloshyna) [inline src/utils/showActionMenu.js:10]:
```suggestion
  const hasPerm = stripes.hasPerm(permissions.SETTINGS_MANAGE);
  
  return hasPerm ? renderer : null;
```

**Review by mariia-aloshyna — APPROVED** [review]

**Pull request merged**